packages/app-builder/src/components/Scenario/AstBuilder/AstBuilderNode/FuzzyMatchComparatorEdit/Modal.tsx

```diff
@@ -0,0 +1,285 @@
+                }
+                onSave={(vm: FuzzyMatchEditorNodeViewModel) => {
+                  const astNode = adaptAstNodeFromEditorViewModel({
+                    ...fuzzyMatchComparatorEditModalProps.initialValue,
+                    children: [
+                      vm,
+                      fuzzyMatchComparatorEditModalProps.initialValue
+                        .children[1],
+                    ],
+                  });
+                  fuzzyMatchComparatorEditModalProps.onSave(astNode);
+                  onOpenChange(false);
+                }}
+              />
+            ) : null}
+          </CopyPasteASTContextProvider>
+        </ModalV2.Content>
+      </FuzzyMatchComparatorEditModalContext.Provider>
+    </ModalV2.Root>
+  );
+}
+
+function FuzzyMatchComparatorEditModalContent({
+  initialValue,
+  onSave,
+}: {
+  initialValue: FuzzyMatchEditorNodeViewModel;
+  onSave: (vm: FuzzyMatchEditorNodeViewModel) => void;
+}) {
+  const { t } = useTranslation(['scenarios', 'common']);
+  const getNodeEvaluationErrorMessage = useGetNodeEvaluationErrorMessage();
+  const [value, setValue] = useState<FuzzyMatchEditorNodeViewModel>(
+    () => initialValue,
+  );
+
+  const handleSave = () => {
+    onSave(value);
+  };
+
+  const options = useOperandOptions({ operandViewModel: value.children[0] });
+  const leftOptions = useMemo(
+    () => options.filter((option) => option.dataType === 'String'),
+    [options],
+  );
+  const rightOptions = useMemo(
+    () =>
+      options.filter(
+        (option) =>
+          option.operandType === 'CustomList' ||
+          option.dataType === 'String[]' ||
+          option.dataType === 'String',
+      ),
+    [options],
+  );
+
+  const adaptEditableAstNode = useAdaptEditableAstNode();
+
+  return (
+    <>
+      <ModalV2.Title>{t('scenarios:edit_fuzzy_match.title')}</ModalV2.Title>
+      <div className="flex flex-col gap-6 p-6">
+        <div className="flex flex-col gap-2">
+          <div className="flex gap-2">
+            <Operand
+              operandViewModel={value.children[0]}
+              onSave={(ast) =>
+                setValue({
+                  ...value,
+                  errors: value.errors.filter(
+                    (error) => error.argumentIndex !== 0,
+                  ),
+                  children: [
+                    adaptEditorNodeViewModel({ ast, parent: value }),
+                    value.children[1],
+                  ],
+                })
+              }
+              options={leftOptions}
+            />
+            <div className="border-grey-10 bg-grey-02 flex h-10 w-fit min-w-[40px] items-center justify-center rounded border p-2 text-center">
+              <span className="text-s text-grey-100 font-medium">~</span>
+            </div>
+            <Operand
+              operandViewModel={value.children[1]}
+              onSave={(ast) => {
+                const nextChild = adaptEditorNodeViewModel({
```

Comment:
why is this "nextChild" ? 🤔

Reply:
The `onSave` we need to pass here is the one responsible for updating the "right" operand (in `left ~ right`).

In order to build this `onSave` specific callback, we have the global `setValue` that require to update the "parent" AST node (not just the `right` operand). I just named it `nextChild` because this is the "future" child after the update, I can change it to `right` or `child`, it's just a habit I add by building some callback where I needed to manipulate "old" or "current" and "future" or "next" value.

Reply:
All right, gotcha.
Maybe `new` rather than `next` ? I was trying to interpret "next" as "are we iterating over something?".
But it's really a detail, you can ignore this if you want